pace-util/pace/util/grid/helper.py

```diff
@@ -0,0 +1,513 @@
+import dataclasses
+from typing import Any, Optional, Sequence
+
+import pace.util
+from pace.dsl import gt4py_utils as utils
+from pace.dsl.stencil import GridIndexing
+from pace.dsl.typing import FloatFieldI, FloatFieldIJ
```

Comment:
This number of imports from `pace.dsl` are setting off alarm bells for me. Is it possible to make storages in this file without `gt4py_utils`? `quantity_wrap` is only used in one file, `dyn_core.py` in fv3core - can it be put there instead?

The typing is a harder question to answer and we can iterate on after. If we want both util and dsl to be able to depend on gt4py, util only doing so for storages and nothing else, then the type hints for storages will need to live in util and get imported into dsl. However, @jdahm has been telling me FloatField is not technically accurate for storages outside of stencils, so these hints themselves might be overhauled (keeping FloatField hints in dsl for use in stencils, and making a new Storage protocol type hint for use outside of stencils like here).

Reply:
Moved `quantity_wrap` to a private function in `dyn_core`. Let's discuss the typing and that can go into the circular dependency PR.

Reply:
Doesn't need to be addressed in this PR, but we also might consider putting all the dataclass containers in a file named to describe that that's what's inside it. I'm not sure what to expect from `grid.helper`.